src/Microsoft.OData.Core/UriParser/Binders/SelectExpandClauseFinisher.cs

```diff
@@ -20,19 +20,24 @@ internal sealed class SelectExpandClauseFinisher
         /// <param name="clause">the select expand clause to modify.</param>
         public static void AddExplicitNavPropLinksWhereNecessary(SelectExpandClause clause)
         {
-            IEnumerable<SelectItem> selectItems = clause.SelectedItems;
+            // Since we're going to be modifying the clause's selected items we need to make a copy here to avoid failures related to
+            // enumerating a changed collection.
+            IEnumerable<SelectItem> selectItems = clause.SelectedItems.ToList();

```

Comment:
why don't use "IList<SelectItem>"

Reply:
I can update to be an `IList<SelectItem>`. It wasn't a conscious decision, I simply added `ToList()` to the current `IEnumerable<SelectItem>` to force it to materialize. It may be I'll need to update this anyway if `SelectedItems` ends up returning an `IList<SelectItem>` itself (see previous comment).